src/notation/internal/masternotation.cpp

```diff
@@ -417,6 +417,12 @@ mu::ValNt<bool> MasterNotation::needSave() const
 
 void MasterNotation::initExcerpts(const ExcerptNotationList& excerpts)
 {
+    // Scores that are closed may have never been laid out, so we lay them out now
+    for (auto score : masterScore()->scoreList()) {
```

Comment:
Unfortunately, this change is incorrect. Now we won't layout scores that haven't been opened yet. Let's revert this particular change to your previous version. I apologize for misleading you :(

Reply:
Actually, for me it seems to work correctly with your first suggestion, as it is now (see video for example, I export the Soprano part without opening it, and it is laid out correctly). In which case exactly do think this will not work?

https://user-images.githubusercontent.com/93707756/196369716-36c9f732-b2f8-4411-8362-eb6ef96914d0.mp4

Reply:
>In which case exactly do think this will not work?

Just create a score containing several instruments (don't open any of them) and then try to export some of them. None of these instruments will be in masterScore()->scoreList(), so we won't call doLayout() for them

Reply:
Ah! Ok, got it. Now I've tried testing (using the latest version of this PR) a worst-case scenario: score is initially empty, instruments are added from the instruments panel, file has never been saved, and parts have never been opened. When exporting all parts to pdf, they are all correctly laid out (which means doLayout() was called). So I think we're good now.

https://user-images.githubusercontent.com/93707756/196901546-302abd8d-91a5-4582-bfbc-3fcef235f032.mp4